thinc/backends/_custom_kernels.py

```diff
@@ -124,88 +125,101 @@ def seq2col(X, nW, *, lengths=None, out=None, threads_per_block=128, num_blocks=
     return out
 
 
-def maxout(X, out=None, threads_per_block=128, num_blocks=128):
+def maxout(X, threads_per_block=128, num_blocks=128):
```

Comment:
I wonder whether we should define most of these functions with keyword-only arguments, in this case it'd be
```
maxout(X, *, threads_per_block=128, num_blocks=128):
```
Things would break (if at all) in a more straightforward way if we change the list of arguments like this PR does.

(To be clear: I'm not saying things do break or that there's an issue with this PR, but it's just something that came to mind seeing the arguments getting changed for many of these functions. It feels like we probably always want to call these functions with named arguments.)

Reply:
I agree. Especially since some of these functions (like the backprop kernels) take multiple arguments that are tensors with the same shape, it's very easy to accidentally make an error and only notice it when training models. I'll make this my todo item after the generic kernel PR.

Reply:
See https://github.com/explosion/thinc/pull/621